### ISSUE TITLE
Should a click on a MetroTabItems close button really clear the buttons Command and CommandParameter properties?

### ISSUE THREAD

**Issue by lagwagon667** [MahApps/MahApps.Metro#2462]
## What steps will reproduce this issue?

Use `MetroTabControl` and add `MetroTabItems` with `CloseButtonEnabled`set to `true` and `CloseTabCommand` (of the `MetroTabItem`) bound to a command that triggers a dialog (_"are you sure you want to close...blablabla?"_). When denied, the command does not remove the tab, but `closeButton_Click` in `MetroTabItem.cs` sets the `CloseTabCommand` and `CloseTabCommandParameter` properties to null, preventing subsequent command invocations.

### Expected outcome

In my opinion these two properties should remain set to their respective Command and Parameter or am I using this control not in the intended way?

--
### Environment

- MahApps.Metro 1.2.4 as well as on develop and master branches
- Windows 7
- Visual Studio 2015
- .NET Framework 4.6

**Comment by punker76** [comment]
@lagwagon667 Just looked at this and think that we maybe doesn't need that. The fix should be checking the CanExecute to false and don't execute the code behind that.

**Comment by lagwagon667** [comment]
I don't think checking `CanExecute` will be sufficient for above use case. If you want to ask the user whether she really wants to close the tab, you can only do this in the `Execute` method. Why do you remove the references to the command and its parameter anyway?

**Comment by punker76** [comment]
@lagwagon667 I don't know, it's old code, must look at this...

**Comment by punker76** [comment]
@lagwagon667 I fix clearing the commands (I understood now why we does it). But a note, the commands on the MetroTabItem itself are not a good place for doing this what you want. It's better you use the `TabItemClosingEvent` on the MetroTabControl. You can set the Cancel to true if you don't want closing the MetroTabItem.

Hope this helps!

**Comment by punker76** [comment]
@lagwagon667 I put a refactoring note on my todo list for v2.0, cause I don't like what we do with this command closing stuff.

**Comment by bboxstart** [comment]
I just encountered this behaviour this week. Using the event is no solution when applying MVVM. However, I was able to work-around this by using the CloseTabCommand on MetroTabControl instead of the same parameter on MetroTabItem. You then still can use the CloseTabCommandParameter on the MetroTabItem.

However, I do support @lagwagon667 on:
- Using CanExecuteCommand is not a solution when you would like the user to ask for confirmation
- The behaviour as is when using CloseTabCommand of MetroTabItem is not intuïtive.
